src/mappings/utils.ts

```diff
@@ -288,6 +309,61 @@ export async function updateTotalDelegation(
   await delegator.save();
 }
 
+export async function updateIndexerCapacity(
+  address: string,
+  event: FrontierEvmEvent
+): Promise<void> {
+  const indexer = await Indexer.get(address);
+  const delegationId = getDelegationId(address, address);
+  const delegation = await Delegation.get(delegationId);
+  const staking = Staking__factory.connect(
+    STAKING_ADDRESS,
+    new FrontierEthProvider()
+  );
+  const eraManager = EraManager__factory.connect(
+    ERA_MANAGER_ADDRESS,
+    new FrontierEthProvider()
+  );
+
+  const leverageLimit = await staking.indexerLeverageLimit();
+
+  if (indexer) {
+    const indexerStake = delegation?.amount;
+    const indexerTotalStake = indexer?.totalStake;
+
+    const stakeCurr = BigNumber.from(indexerStake?.value.value ?? 0);
```

Comment:
Can have a helper function `bigNumberFrom(value: any): BigNumber`, when the value is undefined, can return BN(0).